File: leet_easy_java/leet_121/leet_121a.java
class Solution {
    public int maxProfit(int[] prices) {
        int len = prices.length, _min = Integer.MAX_VALUE, _max = Integer.MIN_VALUE, profit = 0;
        for(int i=0; i< len; i++){
            int prev_min = _min;
            _min = Math.min(_min, prices[i]);
            _max = Math.max(_max, prices[i]);
            if(prev_min == _min) 
                profit = Math.max(profit, _max - _min);
            else
                _max = Integer.MIN_VALUE;
        }
        return profit;
    }
}

File: leet_easy_javascript/leet_121/leet_121a.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var profit = 0, val = prices[0];
    for(var i=1; i<prices.length; i++){
        if(val <= prices[i]) profit = Math.max(profit, prices[i] - val);
        else val = prices[i];           
    }
    return profit;
};